client/components/domains/connect-domain-step/index.jsx

```diff
@@ -42,6 +55,10 @@ function ConnectDomainStep( { domain, selectedSite, initialSetupInfo, initialSte
 		}
 	}, [ initialStep, setPageSlug ] );
 
+	useEffect( () => {
+		if ( isDomainConnected ) page( domainUseMyDomain( selectedSite.slug ) );
```

Comment:
Very nitpicky: I prefer to use braces even for one liners 🙃 I thought the linter used that rule 🤔 but this is definitely not a blocker